Make testgrid config external

krzyzacy opened the pull request:
Added config.proto in proto3
Added config.yaml as external testgrid config
Added yaml2proto to convert the yaml file to config protobuf

User can edit the config.yaml and generate a new config to testgrid

Todos:
- Need to automatically upload the new config to storage, but that will be in a separate PR

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubernetes/test-infra/430)
<!-- Reviewable:end -->

krzyzacy commented:
test site : https://sen-lu-test.appspot.com/google-gke#gke-reboot-release-1.3

rmmh commented:
Reviewed 1 of 42 files at r1.
Review status: 1 of 42 files reviewed at latest revision, 3 unresolved discussions.

---

*[testgrid/config/config.yaml, line 4 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPyH0iMNbHbEV1eik3M:-KPyH0iMNbHbEV1eik3N:bbxsi8c) ([raw file](https://github.com/kubernetes/test-infra/blob/e5431e1ea2e16cb406911be077e087498d35f90e/testgrid/config/config.yaml#L4)):*
> ```YAML
> - name: kubernetes-kubemark-500-gce
>   query: kubernetes-jenkins/logs/kubernetes-kubemark-500-gce
>   days_of_results: 14
> ```

There's a lot of duplication here. Can you make some "default test group" that will get folded into all the others? It looks like only "name" and "query" are changing between each of these entries-- but it's very hard to see!

---

*[testgrid/config/config.yaml, line 1438 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPyHKQeaIr99o4EC4kJ:-KPyHKQfcqhXU-UkPA-E:b-kvhduz) ([raw file](https://github.com/kubernetes/test-infra/blob/e5431e1ea2e16cb406911be077e087498d35f90e/testgrid/config/config.yaml#L1438)):*
> ```YAML
>   - name: aws
>     test_group_name: kubernetes-e2e-aws
>     open_test_template:
> ```

Ditto for these dashboard_tabs. All the duplication will make it *very* annoying to change the templates. Please make some default dashboard tab that the others will inherit from.

---

*[testgrid/config/yaml2proto/yaml2proto.go, line 28 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPyHdaNOAQwmJQt3xOK:-KPyHdaNOAQwmJQt3xOL:bojaqyc) ([raw file](https://github.com/kubernetes/test-infra/blob/e5431e1ea2e16cb406911be077e087498d35f90e/testgrid/config/yaml2proto/yaml2proto.go#L28)):*
> ```Go
> )
> 
> func Yaml2Proto(inputFile string, outputFile string) error {
> ```

This should take a Reader and a Writer, not two file paths. The caller can arrange for the reader/writer to point to the appropriate paths. (This also makes testing easier).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->

krzyzacy commented:
*[testgrid/config/config.yaml, line 4 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPyH0iMNbHbEV1eik3M:-KPyxicRmU4A8PMaYyEB:boy082p) ([raw file](https://github.com/kubernetes/test-infra/blob/e5431e1ea2e16cb406911be077e087498d35f90e/testgrid/config/config.yaml#L4)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary>
> There's a lot of duplication here. Can you make some "default test group" that will get folded into all the others? It looks like only "name" and "query" are changing between each of these entries-- but it's very hard to see!
</details>
config.yaml updated

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->

krzyzacy commented:
*[testgrid/config/config.yaml, line 1438 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPyHKQeaIr99o4EC4kJ:-KPyxzbAdEwLT_uqU3Jb:bgudp3d) ([raw file](https://github.com/kubernetes/test-infra/blob/e5431e1ea2e16cb406911be077e087498d35f90e/testgrid/config/config.yaml#L1438)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary>
> Ditto for these dashboard_tabs. All the duplication will make it *very* annoying to change the templates. Please make some default dashboard tab that the others will inherit from.
</details>
config.yaml updated

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->

rmmh commented on testgrid/config/config.yaml:
```diff
@@ -0,0 +1,1193 @@
+test_group_default: &DEFAULT_TEST_GROUP
+  name: kubernetes-test-default
```
Why include name/query here? Ideally you'd get an error if you inherited from the default and didn't override name/query.

krzyzacy commented:
Review status: 0 of 46 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[testgrid/config/config.yaml, line 2 \[r2\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2s-r2-2:-KQ32V6yANCjRe100Gng:bss4a2b) ([raw file](https://github.com/kubernetes/test-infra/blob/166be38027e5c8f6a45a2390d95b8bc29d705baf/testgrid/config/config.yaml#L2)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary>
> Why include name/query here? Ideally you'd get an error if you inherited from the default and didn't override name/query.
</details>
Removed.

---

*[testgrid/config/yaml2proto/yaml2proto.go, line 28 \[r1\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPyHdaNOAQwmJQt3xOK:-KQ32GOHrOh4S6mrE1Rt:batmdll) ([raw file](https://github.com/kubernetes/test-infra/blob/e5431e1ea2e16cb406911be077e087498d35f90e/testgrid/config/yaml2proto/yaml2proto.go#L28)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary>
> This should take a Reader and a Writer, not two file paths. The caller can arrange for the reader/writer to point to the appropriate paths. (This also makes testing easier).
</details>
Updated function to take in data bytes rather than filename string

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->

krzyzacy commented:
I Updated the PR and have fixes for all existing comments, please review again.

fejta commented on testgrid/config/config.proto:
```diff
@@ -0,0 +1,109 @@
+}
+
+// A service configuration consisting of multiple test groups and dashboards.
+message Configuration {
```
What do you think about:

1) add a config field for the default, aka `TestGroup default_test_group = 3`
2) use this value to add missing fields from each entry in test_groups
3) Delete all the `- <<: *DEFAULT_TEST_GROUP` lines
4) do the same for the dashboards

This has two benefits:

1) It reduces the size of config.yaml by roughly a third
2) We no longer ask people to learn complex yaml syntax (anchors) in order to understand the testgrid config.

krzyzacy replied:
Sure, do the work from parser rather than in yaml I assume?

fejta replied:
yup!

krzyzacy commented:
Updated again, now config.proto has default_test_group and default_dashboard_tab.
Yaml2Proto will fill up any unset fields ( except for name and gcs_prefix )
And on top of the config.yaml will be the default testgroup and dashboardtab

fejta commented on testgrid/config/config.proto:
```diff
@@ -0,0 +1,113 @@
+  int32 days_of_results = 3;
+
+  enum TestsName {
+	  TESTS_NAME_MIN = 0;
```
Please fix indentation

fejta commented on testgrid/config/config.yaml:
```diff
@@ -0,0 +1,809 @@
+    url: https://k8s-gubernator.appspot.com/builds/<gcs_prefix>
+  code_search_url_template:
+    url: https://github.com/kubernetes/kubernetes/compare/<start-custom-0>...<end-custom-0>
+test_groups:
```
This looks fantastic! Will you please add some comments teaching people what default_test_group, default_dashboard_tab and test_groups do and how to configure them?

Also please add some whitespace between big sections.

fejta commented on testgrid/config/yaml2proto/testdata/IsExternalFalse.yaml:
```diff
@@ -0,0 +1,32 @@
+default_test_group:
```
Let's use a consistent naming pattern: other areas of this repo (and org) use hyphens to separate out words rather than camel case.

https://github.com/kubernetes/test-infra/tree/master/jenkins/job-configs/kubernetes-jenkins

fejta commented on testgrid/config/yaml2proto/testdata/NoTestgroup.yaml:
```diff
@@ -0,0 +1,28 @@
+default_test_group:
```
Please use TestGroup or Testgroup consistently. Given default_test_group I'd expect this to be test-group

fejta commented on testgrid/config/yaml2proto/testdata/large.yaml:
```diff
@@ -0,0 +1,809 @@
+default_test_group:
```
Please stay consistent in your filenames. Make all files lower case or all files capitalized.

fejta commented on testgrid/config/yaml2proto/testdata/large.yaml:
```diff
@@ -0,0 +1,809 @@
+    url: https://k8s-gubernator.appspot.com/builds/<gcs_prefix>
+  code_search_url_template:
+    url: https://github.com/kubernetes/kubernetes/compare/<start-custom-0>...<end-custom-0>
+test_groups:
```
Why do we need this test case? How is it different from others?

fejta commented on testgrid/config/yaml2proto/yaml2proto_test.go:
```diff
@@ -0,0 +1,170 @@
+)
+
+func TestYaml2ProtoEmpty(t *testing.T) {
+	yamlData, err := ioutil.ReadFile("testdata/Empty.yaml")
```
Is there a good reason for Empty.yaml to live in another file rather than just defining it inline here?

fejta commented on testgrid/config/yaml2proto/yaml2proto_test.go:
```diff
@@ -0,0 +1,170 @@
+	pb "k8s.io/test-infra/testgrid/config/pb"
+)
+
+func TestYaml2ProtoEmpty(t *testing.T) {
```
Please read the development guide's preference for table driven testing, which seems relevant here https://github.com/kubernetes/kubernetes/blob/master/docs/devel/testing.md#unit-tests

fejta replied:
Will you ask @apelisse if he thinks these tests are a good candidate for table driven testing?

fejta commented on testgrid/config/yaml2proto/yaml2proto_test.go:
```diff
@@ -0,0 +1,170 @@
+	"io/ioutil"
+
+	"github.com/golang/protobuf/proto"
+	pb "k8s.io/test-infra/testgrid/config/pb"
```
What is the purpose of the pb prefix here?

krzyzacy commented:
*[testgrid/config/config.proto, line 17 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2r-r4-17:-KQNWPZ3UoIXDGWL3ArH:bb7f2mf) ([raw file](https://github.com/kubernetes/test-infra/blob/c1e6ed4ab9d586bd92ee5e4d29d538d9298213cc/testgrid/config/config.proto#L17)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Please fix indentation
</details>
done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->

krzyzacy commented:
Review status: 0 of 41 files reviewed at latest revision, 14 unresolved discussions, some commit checks failed.

---

*[testgrid/config/config.proto, line 102 \[r3\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2r-r3-102:-KQNWShol0rRnPhojps4:b-9t87k9) ([raw file](https://github.com/kubernetes/test-infra/blob/62dcd38974f32f701d854eb5dc8b95449ba22dfb/testgrid/config/config.proto#L102)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> yup!
</details>
rewrote the yaml and default configuration

---

*[testgrid/config/config.yaml, line 24 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2s-r4-24:-KQNWazdql9fk1js0w6R:b802a7t) ([raw file](https://github.com/kubernetes/test-infra/blob/c1e6ed4ab9d586bd92ee5e4d29d538d9298213cc/testgrid/config/config.yaml#L24)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> This looks fantastic! Will you please add some comments teaching people what default_test_group, default_dashboard_tab and test_groups do and how to configure them?
> 
> 
> 
> Also please add some whitespace between big sections.
</details>
Comments added

---

*[testgrid/config/yaml2proto/yaml2proto_test.go, line 24 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2y-r4-24:-KQNWfjT2MbGS7bq1VLo:ba3dfob) ([raw file](https://github.com/kubernetes/test-infra/blob/c1e6ed4ab9d586bd92ee5e4d29d538d9298213cc/testgrid/config/yaml2proto/yaml2proto_test.go#L24)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> What is the purpose of the pb prefix here?
</details>
Removed

---

*[testgrid/config/yaml2proto/yaml2proto_test.go, line 27 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2y-r4-27:-KQNWgQNQxOuuUsSpxTD:bb6rskq) ([raw file](https://github.com/kubernetes/test-infra/blob/c1e6ed4ab9d586bd92ee5e4d29d538d9298213cc/testgrid/config/yaml2proto/yaml2proto_test.go#L27)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Please read the development guide's preference for table driven testing, which seems relevant here https://github.com/kubernetes/kubernetes/blob/master/docs/devel/testing.md#unit-tests
</details>
ACK.

---

*[testgrid/config/yaml2proto/yaml2proto_test.go, line 28 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2y-r4-28:-KQNWiF17f9wHBPGTISc:b-sojvkm) ([raw file](https://github.com/kubernetes/test-infra/blob/c1e6ed4ab9d586bd92ee5e4d29d538d9298213cc/testgrid/config/yaml2proto/yaml2proto_test.go#L28)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Is there a good reason for Empty.yaml to live in another file rather than just defining it inline here?
</details>
All test yaml are inline now

---

*[testgrid/config/yaml2proto/testdata/IsExternalFalse.yaml, line 1 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KQ31bBKCia-NLIZq0bq-r4-1:-KQNWnY5gQ9WwV5FK9s6:b4twodr) ([raw file](https://github.com/kubernetes/test-infra/blob/c1e6ed4ab9d586bd92ee5e4d29d538d9298213cc/testgrid/config/yaml2proto/testdata/IsExternalFalse.yaml#L1)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Let's use a consistent naming pattern: other areas of this repo (and org) use hyphens to separate out words rather than camel case.
> 
> 
> 
> https://github.com/kubernetes/test-infra/tree/master/jenkins/job-configs/kubernetes-jenkins
</details>
Removed those files.

---

*[testgrid/config/yaml2proto/testdata/large.yaml, line 1 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2v-r4-1:-KQNWuAI_KpjYN00LbqX:ba3dfob) ([raw file](https://github.com/kubernetes/test-infra/blob/c1e6ed4ab9d586bd92ee5e4d29d538d9298213cc/testgrid/config/yaml2proto/testdata/large.yaml#L1)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Please stay consistent in your filenames. Make all files lower case or all files capitalized.
</details>
Removed

---

*[testgrid/config/yaml2proto/testdata/large.yaml, line 24 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2v-r4-24:-KQNWukxSJLs0R4EYki1:ba3dfob) ([raw file](https://github.com/kubernetes/test-infra/blob/c1e6ed4ab9d586bd92ee5e4d29d538d9298213cc/testgrid/config/yaml2proto/testdata/large.yaml#L24)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Why do we need this test case? How is it different from others?
</details>
Removed

---

*[testgrid/config/yaml2proto/testdata/NoTestgroup.yaml, line 1 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KQ31bBKCia-NLIZq0bs-r4-1:-KQNWwKYGOkw7-lVNeEh:ba3dfob) ([raw file](https://github.com/kubernetes/test-infra/blob/c1e6ed4ab9d586bd92ee5e4d29d538d9298213cc/testgrid/config/yaml2proto/testdata/NoTestgroup.yaml#L1)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Please use TestGroup or Testgroup consistently. Given default_test_group I'd expect this to be test-group
</details>
Removed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->

krzyzacy commented:
So I updated the PR again, please take another look :)

fejta commented on testgrid/config/config.yaml:
```diff
@@ -0,0 +1,831 @@
+  days_of_results: 14
+  tests_name_policy: 2
+  column_header:
+  - configuration_value: Commit
```
Maybe add a comment explaining what this line does?

fejta commented on testgrid/config/yaml2proto/yaml2proto.go:
```diff
@@ -0,0 +1,128 @@
+		return nil, err
+	}
+
+	err = yaml.Unmarshal(yamlData,&defaultConfig)
```
So Unmarhsal just ignores unrecognized fields?

krzyzacy commented:
Review status: 0 of 41 files reviewed at latest revision, 16 unresolved discussions.

---

*[testgrid/config/yaml2proto/yaml2proto.go, line 94 \[r5\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2x-r5-94:-KQN_k5eJFsWgqNtCLrF:b-lzbfvj) ([raw file](https://github.com/kubernetes/test-infra/blob/7a09be75d80372ef5fe4602dd47cf2b215ca1540/testgrid/config/yaml2proto/yaml2proto.go#L94)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> So Unmarhsal just ignores unrecognized fields?
</details>
Yes since it cannot map unrecognized fields, and I need to call it twice since now we have two top-level messages.

I can split the default one into a new yaml

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->

fejta commented on testgrid/config/config_test.go:
```diff
@@ -0,0 +1,133 @@
+	"k8s.io/test-infra/testgrid/config/pb"
+)
+
+func TestConfig(t *testing.T) {
```
These are some great tests!

fejta commented:
Generally looks good, but will you add the necessary .travis.yaml changes to run these unit tests automatically?

fejta commented on testgrid/config/yaml2proto/yaml2proto_test.go:
```diff
@@ -17,46 +17,15 @@ limitations under the License.
-	"github.com/golang/protobuf/proto"
-	pb "k8s.io/test-infra/testgrid/config/pb"
+    "github.com/golang/protobuf/proto"
+    pb "k8s.io/test-infra/testgrid/config/pb"
```
Remove the pb?

krzyzacy commented:
Review status: 0 of 42 files reviewed at latest revision, 18 unresolved discussions.

---

*[testgrid/config/yaml2proto/yaml2proto_test.go, line 27 \[r4\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2y-r4-27:-KQNyMMU9E7H31tJTqsV:b-bw8e14) ([raw file](https://github.com/kubernetes/test-infra/blob/c1e6ed4ab9d586bd92ee5e4d29d538d9298213cc/testgrid/config/yaml2proto/yaml2proto_test.go#L27)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Will you ask @apelisse if he thinks these tests are a good candidate for table driven testing?
</details>
Updated.

---

*[testgrid/config/yaml2proto/yaml2proto_test.go, line 24 \[r6\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2y-r6-24:-KQNyLU79zBZ9VrhUshT:bb7f2mf) ([raw file](https://github.com/kubernetes/test-infra/blob/c99f0e6d8e7936f69d951d36851bd9f4f4c23fa2/testgrid/config/yaml2proto/yaml2proto_test.go#L24)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Remove the pb?
</details>
done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->

fejta commented:
Please assign back to me once tests pass. Ryan/Joe can help with Travis. Antoine can help with vendoring.

fejta commented:
Reviewed 6 of 42 files at r1, 2 of 16 files at r5, 3 of 4 files at r6, 3 of 4 files at r7.
Review status: 14 of 42 files reviewed at latest revision, 6 unresolved discussions.

---

*[testgrid/config/config.yaml, line 17 \[r5\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2s-r5-17:-KQRgRb3ghF4fCgBgqOi:b-rui52z) ([raw file](https://github.com/kubernetes/test-infra/blob/7a09be75d80372ef5fe4602dd47cf2b215ca1540/testgrid/config/config.yaml#L17)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Maybe add a comment explaining what this line does? 
</details>
nit: Inline comments are easier to read

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->

fejta commented:
LGTM

krzyzacy commented:
Review status: 13 of 42 files reviewed at latest revision, 16 unresolved discussions.

---

*[testgrid/config/config.yaml, line 17 \[r5\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KPoGlKz7WsaeJFCKI2s-r5-17:-KQRkPUVpfHn27FNr_jN:bfe5i8v) ([raw file](https://github.com/kubernetes/test-infra/blob/7a09be75d80372ef5fe4602dd47cf2b215ca1540/testgrid/config/config.yaml#L17)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> nit: Inline comments are easier to read
</details>
changed to inline - I feel like without language detector ( like on gedit ), the comments are still hard to read

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->

rmmh commented:
Review status: 13 of 42 files reviewed at latest revision, 18 unresolved discussions.

---

*[.travis.yml, line 14 \[r8\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KQSGB1sSdIGoy9doNzB:-KQSGB1sSdIGoy9doNzC:b4ugrjh) ([raw file](https://github.com/kubernetes/test-infra/blob/5911d849a6f4ed8261512541b4ef81e054f9a693/.travis.yml#L14)):*
> ```YAML
> install:
>     - mkdir -p $HOME/gopath/src/k8s.io
>     - mv $TRAVIS_BUILD_DIR $HOME/gopath/src/k8s.io/test-infra
> ```

does this actually do anything?

---

*[.travis.yml, line 28 \[r8\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KQSGH28fZc8Vqq4LcYv:-KQSGH28fZc8Vqq4LcYw:b-4u28tq) ([raw file](https://github.com/kubernetes/test-infra/blob/5911d849a6f4ed8261512541b4ef81e054f9a693/.travis.yml#L28)):*
> ```YAML
>     - ./lint.sh
>     - cd ../testgrid/config
>     - go test ./...
> ```

If $GOPATH is correctly configured, you should be able to just do `go test k8s.io/test-infra/...` without the cd.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->

krzyzacy commented:
Review status: 13 of 42 files reviewed at latest revision, 18 unresolved discussions.

---

*[.travis.yml, line 14 \[r8\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KQSGB1sSdIGoy9doNzB:-KQSV5Jp90n7-eOnUADu:bti8v4q) ([raw file](https://github.com/kubernetes/test-infra/blob/5911d849a6f4ed8261512541b4ef81e054f9a693/.travis.yml#L14)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary>
> does this actually do anything?
</details>
These lines are still necessary since the initial $TRAVIS_BUILD_DIR is not under gopath, and symlink cause go test ./... not finding packages

---

*[.travis.yml, line 28 \[r8\]](https://reviewable.io:443/reviews/kubernetes/test-infra/430#-KQSGH28fZc8Vqq4LcYv:-KQSVNY-0uS722H1xvWQ:b-7ql1td) ([raw file](https://github.com/kubernetes/test-infra/blob/5911d849a6f4ed8261512541b4ef81e054f9a693/.travis.yml#L28)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary>
> If $GOPATH is correctly configured, you should be able to just do `go test k8s.io/test-infra/...` without the cd.
</details>
fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/430)*
<!-- Sent from Reviewable.io -->
